app/src/main/java/a3/com/convo/activities/LoginActivity.java

```diff
@@ -136,47 +163,52 @@ protected void getLikedPageInfo(AccessToken access_token) {
                     public void onCompleted(
                             JSONObject json_object,
                             GraphResponse response) {
-
-                        try {
-                            final ParseUser user = ParseUser.getCurrentUser();
-                            // initialize empty likes array
-                            user.put("pageLikes", new ArrayList<String>());
-                            // convert Json object into Json array
-                            JSONArray likes = json_object.getJSONObject("likes").optJSONArray("data");
-
-                            for (int i = 0; i < likes.length(); i++) {
-                                final JSONObject page = likes.optJSONObject(i);
-                                String id = page.optString("id");
-
-                                if (existingPages.containsKey(id)) {
-                                    // page already exists in Parse, so we just get the object id and add it to their likes array
-                                    user.add("pageLikes", existingPages.get(id));
-                                } else {
-                                    // doesn't exist yet, so we add it to the server
-                                    String category = page.optString("category");
-                                    String name = page.optString("name");
-                                    String coverUrl = page.getJSONObject("cover").optString("source");
-                                    String profUrl = page.getJSONObject("picture").getJSONObject("data").optString("url");
-                                    final Page newPage = Page.newInstance(id, name, profUrl, coverUrl, category);
-
-                                    newPage.saveInBackground(new SaveCallback() {
-
-                                        @Override
-                                        public void done(ParseException e) {
-                                            if (e == null) {
-                                                Log.e("LoginActivity", "Create page success");
-                                                user.add("pageLikes", newPage.getObjectId());
-                                                user.saveInBackground();
+                        final ParseUser user = ParseUser.getCurrentUser();
```

Comment:
can user be null?

Reply:
We made sure that we call this function only after the user is signed up or logged in and there is no logout button for them to click on. Still, I'll check for that just in case they were somehow automatically logged out of the Parse server.